### ISSUE TITLE
Key-lookup pull queries fail if underlying messages have redundant key fields

### ISSUE THREAD

**Issue by AlanConfluent** [confluentinc/ksql#8829]
**Describe the bug**
This issue comes from an observed behavior with pull queries failing to return known keys.  You could run a query like one of the following:
```
SELECT * FROM INPUT_TABLE;
SELECT * FROM INPUT_TABLE WHERE UNIQUE_ID LIKE 'xxx';
SELECT * FROM INPUT_TABLE WHERE UNIQUE_ID = 'xxxx' EMIT CHANGES;
```
These would return rows, but when you would search for a key using a simple key-based pull query:
```
SELECT * FROM INPUT_TABLE WHERE UNIQUE_ID = 'xxxx';
```
It would find nothing.

**To Reproduce**
To set things up, here were the sources defined:

```
CREATE STREAM INPUT_STREAM (UNIQUE_ID STRING KEY, DATA INTEGER) WITH (KAFKA_TOPIC='input', VALUE_FORMAT='JSON');

CREATE TABLE INPUT_TABLE  AS SELECT
  INPUT_STREAM.UNIQUE_ID UNIQUE_ID,
  LATEST_BY_OFFSET(INPUT_STREAM.DATA) DATA,
  FROM INPUT_STREAM INPUT_STREAM
GROUP BY INPUT_STREAM.UNIQUE_ID;
```
Notice the use of `KEY` in `INPUT_STREAM`. After some investigation, I noticed that the messages being produced to the topic `input` had both a key and a field in the value, e.g.
Key: `unique_id_1234`
Value: `{"unique_id": "unique_id_1234", "data": 1111}` 

Once the messages were changed to remove the redundant `unique_id` in the value body, the above bug was fixed and the messages were queryable in the table, e.g.
Key: `unique_id_1234`
Value: `{"data": 1111}` 

Another solution was to remove the `KEY` keyword in the definition of `INPUT_STREAM`, and then the included `unique_id` in the value message was fine (and expected).

So, to summarize, including the `KEY` in the value body seems to break querying behavior.

**Expected behavior**
You should be able to produce Kafka messages of the form:
Key: `unique_id_1234`
Value: `{"unique_id": "unique_id_1234", "data": 1111}` 

This type of redundancy is pretty common and shouldn't break any behavior, including pull queries.  If `KEY` is used, the redundant field in the value body should be ignored.  If `KEY` is not used, then a key field of the message should be ignored.  

**Actual behaviour**
Broken pull queries.

*Something to also test*: If a table is defined directly on a topic, does the use of `PRIMARY KEY` have a similar effect?

**Additional context**
